buku.py

```diff
@@ -2454,7 +2561,14 @@ def main():
         if len(keywords) > 1:
             tags = parse_tags(keywords[1:])
 
-        bdb.add_rec(args.add[0], title_in, tags, desc_in, args.immutable)
+        url = args.add[0]
+        if args.editor is not None:
+            result = open_editor(args.editor, url, title_in, tags, desc_in)
```

Comment:
Probably we can have a separate API to check editor is available or not and call that from add and update option processing from inside main().